shell.nix

```diff
@@ -2,24 +2,68 @@
   pkgs ? import <nixpkgs> { },
 }:
 
-pkgs.mkShell {
-  inputsFrom = [
-    pkgs.ladybird
-  ];
+(pkgs.buildFHSEnv {
+  name = "nix-shell";
 
-  packages =
+  targetPkgs = pkgs:
     with pkgs;
-    with pkgs.qt6Packages;
-    with pkgs.nodePackages;
     [
-      qtbase.dev
-      qttools
-      qtwayland.dev
+      cmake
+      ninja
+      pkg-config
+      python3
+
+      curl
+      ffmpeg
+      fontconfig
+      libavif
+      libGL
+      libjxl
+      libwebp
+      libxcrypt
+      openssl
+      qt6Packages.qtbase
+      qt6Packages.qtmultimedia
+      simdutf
+      (skia.overrideAttrs (prev: {
+        gnFlags = prev.gnFlags ++ [
+          # https://github.com/LadybirdBrowser/ladybird/commit/af3d46dc06829dad65309306be5ea6fbc6a587ec
+          # https://github.com/LadybirdBrowser/ladybird/commit/4d7b7178f9d50fff97101ea18277ebc9b60e2c7c
+          # Remove when/if this gets upstreamed in skia.
+          "extra_cflags+=[\"-DSKCMS_API=__attribute__((visibility(\\\"default\\\")))\"]"
+        ];
+      }))
+      woff2
+
+      qt6Packages.qtbase.dev
+      qt6Packages.qttools
+      qt6Packages.qtwayland.dev
 
       ccache
       clang-tools
+      nodePackages.prettier
       pre-commit
-      prettier
+
+      autoconf
+      autoconf-archive
+      automake
+      curl
+      ffmpeg.dev
+      glibc.dev
+      gnutar
+      libglvnd.dev
+      nasm
+      unzip
+      xorg.libX11.dev
+      xorg.xorgproto
+      zip
+
+      # Linux-specific
```

Comment:
seems a bunch more of these are "linux specific" tho, right? 

qt6Packages.qtwayland.dev
glibc.dev
libglvnd.dev
xorg.*

Reply:
oh hm. perhaps those other ones already have no-op shims for non-linux.